core/src/main/java/org/apache/accumulo/core/spi/compaction/DefaultCompactionPlanner.java

```diff
@@ -146,15 +172,25 @@ public String toString() {
       justification = "Field is written by Gson")
   @Override
   public void init(InitParameters params) {
-    ExecutorConfig[] execConfigs =
-        GSON.get().fromJson(params.getOptions().get("executors"), ExecutorConfig[].class);
-
     List<Executor> tmpExec = new ArrayList<>();
+    ExecutorConfig[] execConfigs = null;
+    QueueConfig[] queueConfigs = null;
+
+    try {
+      execConfigs =
+          GSON.get().fromJson(params.getOptions().get("executors"), ExecutorConfig[].class);
+    } catch (NullPointerException npe) {
+      // npe could result from executors not being set as properties.
+    } finally {
+      // Generated a zero-length array to avoid a npe thrown by forEach
+      if (execConfigs == null) {
+        execConfigs = new ExecutorConfig[0];
+      }
+    }
```

Comment:
Could do something like the following to avoid catching the NPE

```suggestion
    if(params.getOptions().containsKey("executors)){
      execConfigs =
            GSON.get().fromJson(params.getOptions().get("executors"), ExecutorConfig[].class);
    }else{
      execConfigs = new ExecutorConfig[0];
    }
```

Reply:
Had to account for some other blank/empty string conditions but I was able to rework the logic to avoid these try catch statements.